source/elements/oneDAL/source/algorithms/decomposition/pca.rst

```diff
@@ -68,397 +93,105 @@ signum function,
    arbitrary technique that modifies the eigenvectors' signs.
 
 
--------------
-Usage example
--------------
-::
-
-   oneapi::dal::pca::model run_training(const oneapi::dal::table& data) {
-
-      const auto pca_desc = oneapi::dal::pca::desc<float>{}
-         .set_component_count(5)
-         .set_deterministic(true);
-
-      const auto result = oneapi::dal::train(pca_desc, data);
-
-      print_table("means", result.get_means());
-      print_table("variances", result.get_variances());
-      print_table("eigenvalues", result.get_eigenvalues());
-      print_table("eigenvectors", result.get_model().get_eigenvectors());
-
-      return result.get_model();
-   }
-
-
-::
-
-   oneapi::dal::table run_inference(const oneapi::dal::pca::model& model,
-                               const oneapi::dal::table& new_data) {
-
-      const auto pca_desc = oneapi::dal::pca::desc<float>{}
-         .set_component_count(model.get_component_count());
+.. _pca_i_math:
 
-      const auto result = oneapi::dal::infer(pca_desc, model, new_data);
+Inference
+---------
+Given the inference set :math:`X' = \{ x_1', \ldots, x_m' \}` of
+:math:`p`-dimensional feature vectors and the :math:`r \times p` matrix
+:math:`T` produced at the training stage, the problem is to transform :math:`X'`
+to the set :math:`X'' = \{ x_1'', \ldots, x_m'' \}`, where :math:`x_{j}''` is
+:math:`r`-dimensional feature vector, :math:`1 \leq j \leq m`.
 
-      print_table("labels", result.get_transformed_data());
-   }
+The feature vector :math:`x_{j}''` is computed by applying linear transformation
+[Lang87]_ with the matrix :math:`T` to the feature vector :math:`x_{j}'`,
 
+.. math::
+   :label: x_transform
 
----
-API
----
-Methods
--------
-::
-
-   namespace method {
-      struct cov {};
-      struct svd {};
-      using by_default = cov;
-   } // namespace method
-
-
-.. namespace:: oneapi::dal::pca::method
-
-.. struct:: cov
-
-   Tag-type that denotes `Covariance-based method`_.
+   x_{j}'' = T x_{j}', \quad 1 \leq j \leq m.
 
 
-.. struct:: svd
+.. _pca_i_math_cov:
+.. _pca_i_math_svd:
 
-   Tag-type that denotes `SVD-based method`_.
+Inference methods: *Covariance* and *SVD*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Covariance and SVD inference methods compute :math:`x_{j}''` according to
+:eq:`x_transform`.
 
 
-.. type:: by_default = cov
+-------------
+Usage example
+-------------
+Training
+--------
+.. onedal_code:: oneapi::dal::pca::example::run_training
 
-   Alias tag-type for the `Covariance-based method`_.
+Inference
+---------
+.. onedal_code:: oneapi::dal::pca::example::run_inference
```

Comment:
You are using `print_table()` service function. Is it documented somewhere?

Reply:
Not yet, will be part of other PR if time permits